cli/command/system/info.go

```diff
@@ -207,9 +225,24 @@ func prettyPrintInfo(dockerCli command.Cli, info types.Info) error {
 	if info.ProductLicense != "" {
 		fmt.Fprintln(dockerCli.Out(), "Product License:", info.ProductLicense)
 	}
+	if len(info.CLIPlugins) > 0 {
+		fmt.Fprintln(dockerCli.Out(), "CLI Plugins:")
+		fmt.Fprintln(dockerCli.Out(), " valid:")
+		for _, p := range info.CLIPlugins {
+			if p.Err == nil {
+				fmt.Fprintf(dockerCli.Out(), "  %s: (%s, %s) %s\n", p.Name, p.Version, p.Vendor, p.ShortDescription)
+			}
+		}
+		fmt.Fprintln(dockerCli.Out(), " invalid:")
```

Comment:
Bit on the fence on the split between `valid` and `invalid`. Thinking if we can show them in the same list, given that we already print the error.

Actually; perhaps we should not list the invalid ones here at all, and add a warning for them to `info.Warnings`. WDYT?

Reply:
Yes, I think putting it in the Warnings would be a great way to make them stand out without complicating this output. Invalid plugins would still be listed in the `--format {{.json}}`


Oh; just noticed this (and perhaps I should fix that for the Warnings as well); if there's no plugins, the value is `null`;

```json
{
  "Warnings": null,
  "CLIPlugins": null
}
```

Perhaps we should do a `Mirrors = make([]string, 0)` in case it's `nil`, so that it would become;

```json
{
  "Warnings": null,
  "CLIPlugins": []
}
```

Reply:
@thaJeztah Do you prefer `[]` to tagging it `omitempty` and skipping it altogether?

Reply:
I should check, but I think having `[]` helps people that use the `--format` option; i.e. `--format '{{.CLIPlugins}}'` not producing an error if no plugin happens to be installed.

Reply:
Is there some trick to this?

In the `--format {{json .}}` output I'm getting `"Plugins":null` whether or not I include a 
```go
	if info.ClientInfo.Plugins == nil {
		info.ClientInfo.Plugins = make([]pluginmanager.Plugin, 0)
	}
```
(also tried `if len(...) == 0`). And `-f '{{.ClientInfo.Plugins}}'` works too no matter what.

Reply:
Nevermind, I was looking in the wrong place, adding the `make` *is* fixing the JSON.

I don't think `-f {{.ClientInfo.Plugins}}` was broken in the way you implied though, but it seems to work so that's fine by me ;-)